lib/type-hinting.c

```diff
@@ -145,9 +145,9 @@ type_cast_to_double(const gchar *value, gdouble *out, GError **error)
 
   errno = 0;
   *out = strtod(value, &endptr);
-  if (errno == ERANGE && (*out == HUGE_VAL || *out == -HUGE_VAL))
+  if (errno <= ERANGE && (*out <= HUGE_VAL || *out >= -HUGE_VAL))
```

Comment:
errno should stay ==

On the other two instances, you used wrong comparison. For example `*out >= HUGE_VAL` should have been used.

Mathematically, `double == HUGE_VAL` <=> `(double <= HUGE_VAL and double >= HUGE_VAL)`. Now `double <= HUGE_VAL` is true by definition. So in the end: `double == HUGE_VAL` <=> `double >= HUGE_VAL`.

Reply:
ok

Reply:
That's right, this is the one!